src/ui/controller.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+CKEDITOR.define( [
+	'collection',
+	'model',
+	'ckeditorerror',
+], function( Collection, Model, CKEditorError ) {
+	class Controller extends Model {
+		/**
+		 * Creates an instance of the {@link Controller} class.
+		 *
+		 * @param {Model} [model] Model of this Controller.
+		 * @param {View} [view] View instance of this Controller.
+		 * @constructor
+		 */
+		constructor( model, view ) {
+			super();
+
+			/**
+			 * Model of this controller.
+			 *
+			 * @property {Model}
+			 */
+			this.model = model || null;
+
+			/**
+			 * Set `true` after {@link #init}.
+			 *
+			 * @property {Boolean}
+			 */
+			this.ready = false;
+
+			/**
+			 * View of this controller.
+			 *
+			 * @property {View}
+			 */
+			this.view = view || null;
+
+			/**
+			 * A collection of {@link ControllerCollection} instances containing
+			 * child controllers.
+			 *
+			 * @property {Collection}
+			 */
+			this.collections = new Collection( {
```

Comment:
Can't we call them `children`? I know that items of this collection are not children yet, but it won't look that bad:

```
collection.children.get( 'buttons' ).get( 0 );
```